csharp/extractor/Semmle.Extraction.CSharp/Entities/Expressions/ArrayCreation.cs

```diff
@@ -41,7 +42,7 @@ protected override void Populate()
 
                         var info = new ExpressionInfo(
                             cx,
-                            Type.Create(cx, cx.Compilation.GetSpecialType(Microsoft.CodeAnalysis.SpecialType.System_Int32)),
+                            new AnnotatedType(Entities.Type.Create(cx, cx.Compilation.GetSpecialType(Microsoft.CodeAnalysis.SpecialType.System_Int32)), NullableAnnotation.NotAnnotated),
```

Comment:
Should we add a constructor to `ExpressionInfo` for the case where there is no annotation? I.e.
```
public ExpressionInfo(Context cx, Type type, Extraction.Entities.Location location, ExprKind kind, IExpressionParentEntity parent, int child, bool isCompilerGenerated, string value) : this(cx, new AnnotatedType(type, NullableAnnotation.NotAnnotated), location, kind, parent, child, isCompilerGenerated, value) { }
```

Reply:
I have a preference for it to be explicit, because otherwise a developer could accidentally discard the annotations.

Reply:
Makes sense.